### PR TITLE
Fetch api

### PR DESCRIPTION
**Pull request by kiw808**
- Data is fetched from the [PokeAPIi](https://pokeapi.co/)
- Types are moved to their own files

### CODE REVIEW

**Review by florentbarriol — APPROVED** [review]

**Review by lbonanni — APPROVED** [review]

**Comment by kiw808** [comment]
Looking at the network tab of dev tools, it seems that I have an infinite loop of API requests, but cannot find how to solve it

**Comment by kiw808** [comment]
Had to use `useEffect0` to run the request only once

File: src/App.res [hunk 1]
```diff
@@ -2,7 +2,7 @@
 let make = () => {
   let (pokemonList: option<PokemonList.t>, setPokemonList) = React.useState(() => None)
 
-  React.useEffect(() => {
+  React.useEffect0(() => {
```
Comment (WhyThat) [inline src/App.res]: You can use an `useEffect1` with `setPokemonList` in the dependencies

Comment (kiw808) [reply src/App.res]: Yes, seems more accurate, thanks :)

**Pull request merged**